javalib/src/main/scala/java/lang/Character.scala

```diff
@@ -1371,189 +1371,433 @@ object Character {
   }
 
   object UnicodeBlock {
-    val SURROGATES_AREA    = new UnicodeBlock("SURROGATES_AREA", 0x0, 0x0)
-    val BASIC_LATIN        = new UnicodeBlock("BASIC_LATIN", 0x0, 0x7f)
-    val LATIN_1_SUPPLEMENT = new UnicodeBlock("LATIN_1_SUPPLEMENT", 0x80, 0xff)
-    val LATIN_EXTENDED_A   = new UnicodeBlock("LATIN_EXTENDED_A", 0x100, 0x17f)
-    val LATIN_EXTENDED_B   = new UnicodeBlock("LATIN_EXTENDED_B", 0x180, 0x24f)
-    val IPA_EXTENSIONS     = new UnicodeBlock("IPA_EXTENSIONS", 0x250, 0x2af)
+    //////////////////
+    // Begin Generated
+    //////////////////
+
+    // scalastyle:off line.size.limit
+    val SURROGATES_AREA = new UnicodeBlock("SURROGATES_AREA", 0x0, 0x0)
```

Comment:
I think this takes a close look from an SN compiler expert.  

I believe that the next whack of UnicodeBlock declarations should be "lazy val". Of course, that means that any tables using them should be carefully constructed to be
lazy on any path but a small set of exceedingly common UnicodeBlocks (such as English only).  That is a fast&small chain to slow&complete design.

My concerns are the latency/startup_time when the binary begins to execute.  Since these vals are in an object within an object, I believe they will currently all be allocated and initialized at binary startup.  That is quite a few allocations for data that will almost certainly never be used.

I am pretty certain that changing to "lazy" will reduce startup time & heap requirements. I am not certain if it will slightly increase code size. 

I am aware that the code being replaced used "vals" rather than "lazy vals", I am reluctant to see hundreds of more items slowing up binary startup.

Reply:
I seem to remember being advised when I was doing Unicode work for the re2 regex code to use one dimensional arrays (of structures).  And to be careful to make
maps lazy.

I think it prudent to ask what the recommendation from current SN leadership is.

Reply:
And yes, the re2s and Character.scala unicode data should not be being duplicated.
I think there is at least one pending PR to reduce, not eliminate, some of that
duplication.  IIRC, the idea was to let that PR merge & settle, then do additional 
passes.  I seem to remember to total elimination was not feasible, but that a lot of error-prone duplication could be eliminated.

I believe that known flaw should not keep progress from happening here. I am also open to & welcome better suggestions. Sometimes one simply runs out of time before the next distraction/crises hits and the technical debt remains in the code base for far too long.